etc/compile/compile.sh

```diff
@@ -1,47 +1,46 @@
-#!/bin/sh
+#!/bin/bash
+#
+# This is the first step in building Pachyderm binaries ('pachd' and 'worker')
+# and is called by the 'make docker-build-pachd' and 'make docker-build-worker'
+# make targets. It doesn't actually run the go compiler (that's in
+# run_go_cmds.sh, called at the bottom) but it sets up an unprivileged user
+# with the host caller's user ID, under which the go compiler will run
 
 set -Eex
 
-DIR="$(cd "$(dirname "${0}")/../.." && pwd)"
-cd "${DIR}"
-pwd
+# On OS X, skip setting all of this user/group stuff and just build as root.
+# The way bind mounts work in docker-for-mac are such that even files created
+# by root inside the docker container will be owned by the calling user and
+# group on the host filesystem
 
-BINARY="${1}"
-LD_FLAGS="${2}"
-PROFILE="${3}"
+env
```

Comment:
do we still want this, or was it just for debugging?

Reply:
Jokes, it was for debugging. Now removed.

I stopped printing stdout from the build container without removing this and of course forgot about it.